subworkflows/local/kofamscan.nf

```diff
@@ -0,0 +1,49 @@
+//
+// Run KOFAMSCAN on protein fasta from orf_caller output
+//
+
+include { KOFAMSCAN_SCAN     } from '../../modules/local/kofamscan/scan'
+include { KOFAMSCAN_DOWNLOAD } from '../../modules/local/kofamscan/download'
+include { SUM_KOFAMSCAN      } from '../../modules/local/sum_kofamscan'
+
+workflow KOFAMSCAN {
+
+    take:
+    kofamscan // Channel: val(meta), path(fasta)
+    kofam_dir
+    fcs
+
+    main:
+    ch_versions = Channel.empty()
+
+    String directoryName = kofam_dir
+    File directory       = new File(directoryName)
+    String kofamdb       = directoryName + "/ko_list"
+    File kolistfile      = new File(kofamdb)
+
+    if ( ! directory.exists() ) {
+        directory.mkdir()
+    }
```

Comment:
what do you suggest instead? we have this code here because we wanted to be sure that if we want to download the database, this will be available outside the work directory and it can be reused. we do the same (creating a folder if it doesn't exist) for EUKulele and eggnog but we do it in metatdenovo.nf instead of the subworkflow. should I move this part of the code there?

Reply:
Use the directive `storeDir` for this purpose. https://www.nextflow.io/docs/latest/process.html#storedir

Reply:
Hi, I am little lost in the way I should use `storeDir`. do I need to set it in the subworklow? in the examples in the documentation it looks you can use it inside the processes but I am not sure it's working in the sub workflow. there are any pipeline using this function? I might take a look!

Reply:
I don't see it used in another nf-core repository. However, yes you would set this for the process that downloads the database you want to save. I think in the case of this workflow, it's `KOFAMSCAN_DOWNLOAD` that you set this for. Here's a toy example to play with to see what's going on.

```nextflow
params.long_term_cache_dir = 'cache'

workflow {
    // Make some test files `mkdir data; touch data/test{1..4}.txt`
    test_files = Channel.fromPath('data/test*.txt', checkIfExists: true)
    TASK( test_files.toList() )
}

process TASK {
    input:
    path files, arity: '4'

    storeDir params.long_term_cache_dir

    script:
    """
    mkdir db
    cp $files db
    touch db/db.index
    """

    output:
    path "db", emit: db
}
```
When you run this workflow the first time, you see this (the process is run because the "database" is missing):
```console
$ nextflow run main.nf 
Nextflow 23.10.1 is available - Please consider updating your version to it
N E X T F L O W  ~  version 23.10.0
Launching `main.nf` [ecstatic_waddington] DSL2 - revision: e454c39f16
executor >  local (1)
[ee/dba71f] process > TASK [100%] 1 of 1 ✔
```
but when you run it again (note: without `-resume`), you see the process skipped because the database is in the `storeDir`.
```console
$ nextflow run main.nf 
Nextflow 23.10.1 is available - Please consider updating your version to it
N E X T F L O W  ~  version 23.10.0
Launching `main.nf` [jolly_raman] DSL2 - revision: e454c39f16
[skipped  ] process > TASK [100%] 1 of 1, stored: 1 ✔
[skipping] Stored process > TASK
```

Put the `storeDir` directive in the `modules.config`.

Reply:
but is it allowed from the nf-core guidelines to have a parameter inside a module? I thought it wasn't!

Reply:
Generally no, but this is why I said set this via the `modules.config`. Don't put the directive in the process. That was only to demonstrate it's use without creating a separate config.